src/main/java/com/artipie/rpm/RepoUpdater.java

```diff
@@ -0,0 +1,389 @@
+            throw new IllegalStateException("Could not complete repo update", ex);
+        }
+        return Completable.concatArray(
+            this.primary.complete(),
+            this.filelists.complete(),
+            this.other.complete()
+        ).andThen(
+            Completable.fromAction(
+                () -> {
+                    xml.writeStartDocument("UTF-8", "1.0");
+                    xml.writeStartElement("repomd");
+                    xml.writeDefaultNamespace("http://linux.duke.edu/metadata/repo");
+                    xml.writeStartElement("revision");
+                    // @checkstyle MagicNumberCheck (1 line)
+                    xml.writeCharacters(String.valueOf(System.currentTimeMillis() / 1000L));
+                    xml.writeEndElement();
+                }
+            )
+        ).andThen(
+            this.processType(xml, this.pfile, "primary")
+        ).andThen(
+            this.processType(xml, this.lfile, "filelists")
+        ).andThen(
+            this.processType(xml, this.ofile, "other")
+        ).andThen(
+            Completable.fromAction(
+                () -> {
+                    xml.writeEndElement();
+                    xml.writeEndDocument();
+                    xml.close();
+                }
+            )
+        ).andThen(
+            Single.fromCallable(
+                () -> {
+                    final Transformer transformer = TransformerFactory.newInstance().newTransformer();
+                    transformer.setOutputProperty(OutputKeys.VERSION, "1.0");
+                    transformer.setOutputProperty(OutputKeys.ENCODING, "UTF-8");
+                    transformer.setOutputProperty(OutputKeys.STANDALONE, "no");
+                    transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+                    final StringWriter out = new StringWriter();
+                    transformer.transform(
+                        new StAXSource(
+                            XMLInputFactory.newFactory().createXMLStreamReader(
+                                Files.newInputStream(repomd)
+                            )
+                        ),
+                        new StreamResult(out)
+                    );
+                    return out.toString();
+                }
+            )
+        ).flatMapCompletable(
+            str ->
+                this.stg.save(
+                    new Key.From("repodata/repomd.xml"),
+                    new Content.From(str.getBytes())
+                )
+        ).doOnTerminate(
+            () -> {
+                Files.delete(this.pfile);
+                Files.delete(this.lfile);
+                Files.delete(this.ofile);
+                Files.delete(repomd);
+            }
+        );
+    }
+
+    /**
+     * Deletes old metadata from storage.
+     * @return Completion or error signal
+     */
+    public Completable deleteMetadata() {
+        return this.stg.list(new Key.From("repodata"))
```

Comment:
@brastak it's not correct, the repository is not located in the root of the storage. The location of repo is specified by `key` of `Rpm.batchUpdate()` or it can be the parent of `key` in `Rpm.update()`